flax/core/lift.py

```diff
@@ -1394,3 +1394,63 @@ def inner_loop(scope, carry):
 def _unzip2(xs):
   ys = tuple(zip(*xs))
   return ys if ys else ((), ())
+
+
+def add_metadata_axis(
```

Comment:
Shouldn't this just be a thin wrapper around map_variables?

Reply:
Yeah - I rewrote it.  Not sure why I didn't do that the first time, I was kinda psycho with sleep deprivation at the time.  Just a small helper in transforms.py now.